flake-info/src/bin/flake-info.rs

```diff
@@ -226,25 +237,29 @@ async fn run_command(
                 nixpkgs.channel.to_owned(),
                 nixpkgs.git_ref.to_owned(),
             );
+            // Force --kind package if an attribute was specified
+            let kind = if attribute.is_some() { Kind::Package } else { kind };
 
             Ok((
                 Box::new(move || {
-                    flake_info::process_nixpkgs(&Source::Nixpkgs(nixpkgs), &kind)
+                    flake_info::process_nixpkgs(&Source::Nixpkgs(nixpkgs), &kind, &attribute)
                         .map_err(FlakeInfoError::Nixpkgs)
                 }),
                 ident,
             ))
         }
-        Command::NixpkgsArchive { source, channel } => {
+        Command::NixpkgsArchive { source, channel, attribute } => {
             let ident = (
                 "nixos".to_string(),
                 channel.to_owned(),
                 "latest".to_string(),
             );
+            // Force --kind package if an attribute was specified
+            let kind = if attribute.is_some() { Kind::Package } else { kind };
```

Comment:
should we show an error message when `--attr` is provided but `--kind` specifies options instead?

Reply:
Good idea.